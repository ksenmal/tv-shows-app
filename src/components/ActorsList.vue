<script setup>
import ActorCard from "./ActorCard.vue";

defineProps({
  list: Array,
});
</script>

<template>
  <ul>
    <li v-for="(actor, index) in list" :key="index">
      <ActorCard :actor="actor" />
    </li>
  </ul>
</template>

<style scoped>
ul {
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}
ul li {
  width: 300px;
  display: flex;
  margin-bottom: 2rem;
}
</style>
