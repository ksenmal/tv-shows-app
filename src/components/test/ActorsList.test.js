import { shallowMount } from "@vue/test-utils";
import ActorsList from "../ActorsList.vue";
import { describe, expect, test } from "vitest";

describe("ActorsList suite", () => {
  test("Snapshot when empty", () => {
    const wrapper = shallowMount(ActorsList, {
      props: {
        list: [],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Snapshot with data", () => {
    const wrapper = shallowMount(ActorsList, {
      props: {
        list: [{ id: 1 }, { id: 2 }],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
