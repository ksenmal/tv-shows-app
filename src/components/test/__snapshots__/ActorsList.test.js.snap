// Vitest Snapshot v1

exports[`ActorsList suite > Snapshot when empty 1`] = `"<ul data-v-d83a9334=\\"\\"></ul>"`;

exports[`ActorsList suite > Snapshot with data 1`] = `
"<ul data-v-d83a9334=\\"\\">
  <li data-v-d83a9334=\\"\\">
    <actor-card-stub actor=\\"[object Object]\\" data-v-d83a9334=\\"\\"></actor-card-stub>
  </li>
  <li data-v-d83a9334=\\"\\">
    <actor-card-stub actor=\\"[object Object]\\" data-v-d83a9334=\\"\\"></actor-card-stub>
  </li>
</ul>"
`;
